cms/djangoapps/models/settings/course_metadata.py

```diff
@@ -117,6 +119,10 @@ def filtered_list(cls):
         if not XBlockStudioConfigurationFlag.is_enabled():
             filtered_list.append('allow_unsupported_xblocks')
 
+        # Do not show "Course Visibility For Unenrolled Learners" in Studio Advanced Settings
+        # if the enable_anonymous_access flag is not enabled
+        if not COURSE_ENABLE_UNENROLLED_ACCESS_FLAG.is_enabled(course_key=course_key):
+            filtered_list.append('course_visibility')
```

Comment:
How frequently is this method invoked during courseware rendering? I just want to make sure we're not making a memcached or MySQL call per rendered XBlock to get this course feature flag info.

Reply:
WaffleFlags are stored in the [request cache](https://github.com/edx/edx-platform/blob/master/openedx/core/djangoapps/waffle_utils/__init__.py#L119). But I'll double check this.